### ISSUE TITLE
`pnpm pack` writes unexpected `./` paths in tarball

### ISSUE THREAD

**Issue by hyrious** [pnpm/pnpm#6193]
### Code to reproduce the issue:

```console
mkdir -p test-pnpm-pack/dist
cd test-pnpm-pack
touch dist/index.js
cat <<EOF >package.json
{
  "name": "foo",
  "version": "0.1.0",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ]
}
EOF
pnpm pack
tar tvf foo-0.1.0.tgz
```

Outputs:

```
package/./dist/index.js
package/dist/index.js
package/package.json
```

### Expected behavior:

The row `package/./dist/index.js` should not exist.

### Actual behavior:

`pnpm pack` writes an unwanted entry in the tarball that contains `./` in its name. [You can find a published example here.](https://www.npmjs.com/package/side-effect-manager?activeTab=code)

### Additional information:

 - `node -v` prints: v18.13.0
 - Windows, macOS, or Linux?: Reproduced on Windows 10 and the latest macOS.

**Comment by hyrious** [comment]
This is caused by [npm-packlist@5](https://github.com/pnpm/pnpm/blob/ee61ca4cb7ce6b3cb177f892940edb55b19b9e17/releasing/plugin-commands-publishing/package.json#L77) generates './' entries, which was fixed in npm-packlist@6...

**Comment by NamesMT** [comment]
+1, version: 8.8.0

**Comment by hyrious** [comment]
PNPM seems fixed it in #7250.
